pkg/operator/controller/gatewayclass/subscription.go

```diff
@@ -49,6 +53,13 @@ func (r *reconciler) ensureServiceMeshOperatorSubscription(ctx context.Context)
 
 // desiredSubscription returns the desired subscription.
 func desiredSubscription(name types.NamespacedName) (*operatorsv1alpha1.Subscription, error) {
+	packageName := "servicemeshoperator"
+	catalogSource := "redhat-operators"
+
+	if OKD {
+		packageName = "servicemeshoperator3"
+		catalogSource = "okderators"
+	}
```

Comment:
does this require OKDerators catalog to be installed? what happens if you trigger this feature without installing OKDerators?